src/tasks.ts

```diff
@@ -0,0 +1,102 @@
+import { spawnSync } from "child_process";
+import * as path from "path";
+import { Range, TextEditor } from "vscode";
+import { outputChannel, ws } from "./extension";
+import { TaskOutput } from "./taskRunner";
+import { NameToThemeNumber } from "./themePicker";
+import { util } from "./utility";
+
+/**
+ * D2Tasks - static functions to be used in tasks.  Functions need
+ * to be synchronous, and return their results in a callback.
+ */
+class D2Tasks {
+  public compile(text: string, log?: TaskOutput, err?: TaskOutput): string {
+    const layout: string = ws.get("previewLayout", "dagre");
+    const theme: string = ws.get("previewTheme", "default");
+    const sketch: boolean = ws.get("previewSketch", false);
+    const themeNumber: number = NameToThemeNumber(theme);
+    const d2Path: string = ws.get("execPath", "d2");
+
+    err?.("Starting Compile...");
+    err?.(`Layout: ${layout}  Theme: ${theme}  Sketch: ${sketch}\r\n`);
```

Comment:
```suggestion
    log?.("Starting Compile...");
    log?.(`Layout: ${layout}  Theme: ${theme}  Sketch: ${sketch}\r\n`);
```

Reply:
I changed it to 'terminal' since that's where its going.  So log==outputChannel, terminal==pseudoTerminal.